.github/workflows/aws-deploy.yml

```
@@ -0,0 +1,95 @@
+---
+name: AWS Deployment
+
+on: 
+  workflow_dispatch:
+    inputs:
+        environment:
+            description: "Environment to deploy to"
+            default: "staging"
+            options:
+                - staging
+                - production
+            required: true
+            type: choice
+       
+env:
+  AWS_ACCOUNT_ID: ${{ vars.AWS_PUBLIC_DATA_RELEASES_ACCOUNT_ID }}
+  AWS_REGION: ${{ vars.AWS_DEFAULT_REGION }}
+  STAGING_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.STAGING_CLOUDFRONT_DISTRIBUTION_ID }}
+  STAGING_S3_BUCKET: s3://staging.volumeviewer.allencell.org
+  PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID }}
+  PRODUCTION_S3_BUCKET: s3://volumeviewer.allencell.org
+
+permissions:
+    id-token: write  # Required for requesting the JWT and OIDC
+    contents: write  # Required for actions/checkout and OIDC tokens
+
+jobs:
+    build:
+      runs-on: ubuntu-latest
+
+      steps:
+        - name: Checkout
+          uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
+  
+        - uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8
+          with:
+            node-version: "18"
+  
+        - name: Install and Build
+          run: npm ci && npm run gh-build
+  
+        - name: Upload build files
+          uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3
+          with:
+            name: aws-deploy-files
+            path: ./imageviewer
+    
+    deploy:
+        needs: build
+        runs-on: ubuntu-latest
+
+        # Dynamically set the environment variable based on the input above:
+        environment: ${{ github.event.inputs.environment }}
+
+        steps:
+
+          # Compute a short sha for use in the OIDC session name, which has a 64 character limit
+          - name: Add SHORT_SHA env property with commit short sha
+            run: echo "SHORT_SHA=`echo ${{ github.sha }} | cut -c1-8`" >> $GITHUB_ENV
+
+          - name: Configure AWS credentials with OIDC
+            uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502
+            with:
+                role-to-assume: arn:aws:iam::${{ env.AWS_ACCOUNT_ID }}:role/github_website_3d_cell_viewer
+                role-session-name: website_3d_cell_viewer-${{ env.SHORT_SHA }}
+                aws-region: ${{ env.AWS_REGION }}
+
+          # Setup variables based on the staging or production environment
+          - name: Set ECS variables based on environment
```

Nit: Not an ECS deployment
```suggestion
          - name: Set deployment variables based on environment
```